webapp/components/_new/features/ReportList/ReportList.vue

```diff
@@ -0,0 +1,150 @@
+        {
+          label: this.$t('moderation.reports.filterLabel.unreviewed'),
+          value: this.unreviewedReports,
+        },
+        { label: this.$t('moderation.reports.filterLabel.reviewed'), value: this.reviewedReports },
+        { label: this.$t('moderation.reports.filterLabel.closed'), value: this.closedReports },
+      ]
+    },
+  },
+  methods: {
+    filter(option) {
+      this.reports = option.value
+      this.selected = option.label
+    },
+    confirm(report) {
+      this.openModal(report)
+    },
+    async confirmCallback(resource) {
+      const { disabled: disable, id: resourceId } = resource
+      this.$apollo
+        .mutate({
+          mutation: reviewMutation(),
+          variables: { disable, resourceId, closed: true },
+        })
+        .then(() => {
+          this.$toast.success(this.$t('moderation.reports.DecisionSuccess'))
+          this.$apollo.queries.reportsList.refetch()
+        })
+        .catch(error => this.$toast.error(error.message))
+    },
+    openModal(report) {
+      const identStart =
+        'moderation.reports.decideModal.' +
+        report.resource.__typename +
+        '.' +
+        (report.resource.disabled ? 'disable' : 'enable')
+      this.$store.commit('modal/SET_OPEN', {
+        name: 'confirm',
+        data: {
+          type: report.resource.__typename,
+          resource: report.resource,
+          modalData: {
+            titleIdent: identStart + '.title',
+            messageIdent: identStart + '.message',
+            messageParams: {
+              name:
+                report.resource.__typename === 'User'
+                  ? report.resource.name
+                  : report.resource.__typename === 'Post'
+                  ? this.$filters.truncate(report.resource.title, 30)
+                  : report.resource.__typename === 'Comment'
+                  ? this.$filters.truncate(
+                      this.$filters.removeHtml(report.resource.contentExcerpt),
+                      30,
+                    )
+                  : '',
+            },
+            buttons: {
+              confirm: {
+                danger: true,
+                icon: report.resource.disabled ? 'eye-slash' : 'eye',
+                textIdent: 'moderation.reports.decideModal.submit',
+                callback: () => {
+                  this.confirmCallback(report.resource)
+                },
+              },
+              cancel: {
+                icon: 'close',
+                textIdent: 'moderation.reports.decideModal.cancel',
+                callback: () => {},
+              },
+            },
+          },
+        },
+      })
+    },
+  },
+  apollo: {
+    reportsList: {
+      query: reportsListQuery(),
+      update({ reports }) {
+        this.reports = reports
+        this.allReports = reports
+        this.unreviewedReports = reports.filter(report => !report.reviewed)
```

Comment:
I changed my mind. This is certainly something that we need to `filter` on the backend. Otherwise I see this:

![Screenshot - 2019-11-30T014846 489](https://user-images.githubusercontent.com/2110676/69893575-be864b80-1313-11ea-879c-b47fd08e8ae5.png)

Reply:
not sure how you got into this situation, as I never got here...
did you seed it with the new seed data?
can be that you had unconverted relationships?
I thought about filtering it in the `backend`, but it's challenging because you need to check if a `report` has any `reviewed`... I can check out the filters betters, but one benefit of this way is that we grab one list, filter it and save the values to variables, and we minimize the number of backend calls... 
where should it error out, if the data looks correctly in the database?